### DISCUSSION TITLE
Can we run window server 2019 docker images on window server 2022 host

### DISCUSSION THREAD

**Discussion by sumit84mca** [microsoft/dotnet-framework-docker#1058] [Q&A]
Hi,

I want to run my docker images which are based on  mcr.microsoft.com/dotnet/framework/aspnet:4.8-windowsservercore-ltsc2019 on a window server running window server 2022. 
When I try to create container I get 

`docker: Error response from daemon: hcsshim::CreateComputeSystem b6d87cd03ef9c3f67b7b69cd56462d00b79c46b90c5cf018c83a8e49450829de: The container operating system does not match the host operating system.`

Is it possible to run window server 2019 images on windows server 2022 host.

**Comment by mthalman** [comment]
You can if you use Hyper-V isolation. By default, Docker will use process isolation. You can read more about [Windows container version compatibility](https://learn.microsoft.com/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2022%2Cwindows-11).

Here's the relevant table for you:

![image](https://user-images.githubusercontent.com/15789599/232048477-c9867681-5655-4404-b533-cc7859140b61.png)

Docker has an [`--isolation` option](https://docs.docker.com/engine/reference/commandline/run/#isolation) that let's you configure this.

Note that there are performance considerations when enabling Hyper-V isolation. Read more about [Windows container isolation modes](https://learn.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/hyperv-container).

**Reply by sumit84mca** [reply]
Hi,
I tried using --isolation=hyperv but then I am getting error 
`docker: Error response from daemon: hcsshim::CreateComputeSystem 94b67dd038a11af9f330f07e902280bdeba770e0f949fdc6b2c575a6ba02d847: The virtual machine could not be started because a required feature is not installed.`

Hyper V is already installed. What else is missing..

![image](https://user-images.githubusercontent.com/18229166/232475537-8371f80b-c34e-4c5e-aeae-bcf439e2d2bd.png)
